From f76d15af4d876efe573dad2c9927db4ce3076baf Mon Sep 17 00:00:00 2001
From: Aaron Quint <aaron@quirkey.com>
Date: Sat, 27 Dec 2008 23:29:48 -0500
Subject: [PATCH] Use SimpleLogger for Rails::TemplateRunner outside of the Generator context.

In a previous commit #logger replaced puts in the TemplateRunner this
worked inside of ./script/generate but not for rake rails:template.

This fix should allow it to work in any context.
---
 .../generators/applications/app/template_runner.rb |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/railties/lib/rails_generator/generators/applications/app/template_runner.rb b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
index 4af8576..cb9d9e5 100644
--- a/railties/lib/rails_generator/generators/applications/app/template_runner.rb
+++ b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
@@ -355,5 +355,15 @@ module Rails
     def logger
       @logger ||= Rails::Generator::Base.logger
     end
+    
+    def logger
+      @logger ||= if defined?(Rails::Generator::Base)
+        Rails::Generator::Base.logger
+      else
+        require 'rails_generator/simple_logger'
+        Rails::Generator::SimpleLogger.new(STDOUT)
+      end
+    end
+    
   end
 end
\ No newline at end of file
-- 
1.6.0.2

